modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp

```diff
@@ -609,16 +609,182 @@ class CoreAudioReader final : public AudioFormatReader
     JUCE_DECLARE_NON_COPYABLE_WITH_LEAK_DETECTOR (CoreAudioReader)
 };
 
-//==============================================================================
-CoreAudioFormat::CoreAudioFormat()
-    : AudioFormat (coreAudioFormatName, findFileExtensionsForCoreAudioCodecs()),
-      streamKind (StreamKind::kNone)
+static AudioFormatID formatForFileType (AudioFileTypeID fileType)
 {
+    AudioFormatID formatIds[10];
+    UInt32 sizeOfArray = sizeof (formatIds);
+    AudioFileGetGlobalInfo (
+        kAudioFileGlobalInfo_AvailableFormatIDs, sizeof (fileType), (void*) &fileType, &sizeOfArray, &formatIds);
+    jassert (sizeOfArray != 0);
+    return formatIds[0];
 }
 
+static void fillAudioStreamBasicDescription (AudioStreamBasicDescription* fmt)
+{
+    UInt32 sz = sizeof (AudioStreamBasicDescription);
+    OSStatus e [[maybe_unused]] = AudioFormatGetProperty (kAudioFormatProperty_FormatInfo, 0, nullptr, &sz, fmt);
+    jassertquiet (e == noErr);
+}
+
+class CoreAudioWriter : public AudioFormatWriter
+{
+public:
+    CoreAudioWriter (
+        OutputStream* out, AudioFileTypeID fileType, double sr, unsigned int numberOfChannels, unsigned int bitsPerSamp)
+    : AudioFormatWriter (out, coreAudioFormatName, sr, numberOfChannels, bitsPerSamp)
+    {
+        usesFloatingPointData = true;
+        {
+            AudioStreamBasicDescription fmt;

```

Comment:
This seems to be an incomplete implementation or use of core audio settings. this might be the place to change from AAC LC to AAC HE v2